app/controllers/documents_controller.rb

```diff
@@ -23,48 +23,67 @@ def index
+      presented_links = DocumentLinksPresenter.new(@document)
 
-      item_request = publishing_api.put_content(document.content_id, presented_document.to_json)
-      links_request = publishing_api.put_links(document.content_id, presented_links.to_json)
+      item_request = publishing_api.put_content(@document.content_id, presented_document.to_json)
+      links_request = publishing_api.put_links(@document.content_id, presented_links.to_json)
 
       if item_request.code == 200 && links_request.code == 200
-        flash.now[:success] = "Created #{document.title}"
+        flash.now[:success] = "Created #{@document.title}"
         redirect_to documents_path(current_format.document_type)
       else
-        flash.now[:danger] = "There was an error publishing #{document.title}. Please try again later."
-        render :new, locals: { document: document }
+        flash.now[:danger] = "There was an error creating #{@document.title}. Please try again later."
+        render :new
       end
     else
-      document_errors = document.errors.messages
-      errors = content_tag(:p,
-        %Q{
-          There #{document_errors.length > 1 ? 'were' : 'was' } the following
-          #{document_errors.length > 1 ? 'errors' : 'error' } with your
-          #{current_format.title.singularize}:
-        }
-      )
-      errors += content_tag :ul do
-        document.errors.full_messages.map { |e| content_tag(:li, e) }.join('').html_safe
-      end
-
-      flash.now[:danger] = errors
-      render :new, locals: { document: document }, status: 422
+      flash.now[:danger] = document_error_messages
+      render :new, status: 422
     end
   end
 
   def show
-    @document = current_format.klass.from_publishing_api(publishing_api.get_content(params[:id]).to_ostruct)
+    @document = current_format.klass.from_publishing_api(publishing_api.get_content(params[:content_id]).to_ostruct)
+  end
+
+  def edit
+    @document = current_format.klass.from_publishing_api(publishing_api.get_content(params[:content_id]).to_ostruct)
+  end
+
+  def update
+    @document = current_format.klass.new(
+      filtered_params(params[:"#{current_format.format_name}"]).merge({content_id: params[:content_id]})
+    )
+
+    @document.public_updated_at = Time.zone.now
+
+    if @document.valid?
+      presented_document = DocumentPresenter.new(@document)
+      presented_links = DocumentLinksPresenter.new(@document)
+
+      item_request = publishing_api.put_content(@document.content_id, presented_document.to_json)
+      links_request = publishing_api.put_links(@document.content_id, presented_links.to_json)
+
+      if item_request.code == 200 && links_request.code == 200
+        flash.now[:success] = "Updated #{@document.title}"
+        redirect_to documents_path(current_format.document_type)
+      else
+        flash.now[:danger] = "There was an error updating #{@document.title}. Please try again later."
+        render :new
```

Comment:
I spotted that and accidentally put it in https://github.com/alphagov/specialist-publisher/commit/bca823fc9b4442a70cf6af4c4c0380cf64b3ab5e#diff-a00af01be68875483d702885663e7efaR85. If you want I can move it to this commit?